### PR TITLE
fix DefinePlugin stringifyObj bug

### PR DESCRIPTION
**Pull request by kagawagao**
**What kind of change does this PR introduce?**

bugfix

**Did you add tests for your changes?**

yes

**If relevant, link to documentation update:**
N/A

**Summary**

[issue 4000](https://github.com/webpack/webpack/issues/4000#issuecomment-273479707)

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

**Comment by TheLarkInn** [comment]
Congrats on submitting your first PR.

**Comment by kagawagao** [comment]
🎉  Thanks, I am so excited for this 😄

File: test/configCases/plugins/define-plugin/index.js [hunk 1]
```diff
@@ -75,4 +75,8 @@ it("should define process.env.DEFINED_NESTED_KEY", function() {
 });
 it("should define process.env.DEFINED_NESTED_KEY_STRING", function() {
 	if(process.env.DEFINED_NESTED_KEY_STRING !== "string") require("fail");
-})
+});
+it("should assign to process.env", function() {
+	process.env.TEST = "test";
```
Comment (sokra) [inline test/configCases/plugins/define-plugin/index.js]: This doesn't test your case. Please test if your test fails without your changes.

``` js
var sub = "SUB";
OBJECT[SUB].FUNCTION(1);
```

This test should trigger the wrong behavior.

---

But make sure to also test this case:

``` js
function f() {
  throw new Error("should not be called");
}
f // <- no semicolon here
OBJECT;
```

**Comment by kagawagao** [comment]
@sokra hmm, I overlooked that, it seems like `f` will be called
```javascript
function f() {
  throw new Error("should not be called");
}
f // <- no semicolon here
OBJECT;
```
so I had to use this way
```javascript
function stringifyObj(obj) {
  return "true && ({" + Object.keys(obj).map((key) => {
    let code = obj[key];
    return JSON.stringify(key) + ":" + toCode(code);
  }).join(",") + "})";
}
```
Is there any other better way?

ps. I just take the test case from issue, I did not expect it will not trigger the wrong behavior, sorry for that.

**Pull request merged**